gadgets/Makefile

```diff
@@ -1,24 +1,36 @@
+SHELL := /bin/bash
+
+GADGETS_TAG ?= $(shell ../tools/image-tag branch)
+GADGETS_REPOSITORY ?= ghcr.io/inspektor-gadget
+BUILDER_IMAGE ?= ghcr.io/inspektor-gadget/inspektor-gadget-ebpf-builder:latest
+
 GADGETS = \
-	trace_exec_x86.bpf.o trace_exec_arm64.bpf.o \
-	trace_open_x86.bpf.o trace_open_arm64.bpf.o \
-	trace_tcpconnect_x86.bpf.o trace_tcpconnect_arm64.bpf.o \
-	trace_tcpretrans_x86.bpf.o trace_tcpretrans_arm64.bpf.o \
-	trace_dns_x86.bpf.o trace_dns_arm64.bpf.o \
+	trace_dns \
+	trace_exec \
+	trace_open \
+	trace_tcpconnect \
+	trace_tcpretrans \
 	#
 
 .PHONY: all
 all: $(GADGETS)
 
-%_x86.bpf.o : %.bpf.c
-	clang -target bpf -Wall -g -O2 -D __TARGET_ARCH_x86 -c $(filter %.c,$^) \
-		-I ../include/ \
-		-I ../include/gadget/amd64/ \
-		-o $@
-%_arm64.bpf.o : %.bpf.c
-	clang -target bpf -Wall -g -O2 -D __TARGET_ARCH_arm64 -c $(filter %.c,$^) \
-		-I ../include/ \
-		-I ../include/gadget/arm64/ \
-		-o $@
+#TODO: How to create an independent target for each gadget do this can be run in parallel?
```

Comment:
Suggestion: https://github.com/inspektor-gadget/inspektor-gadget/commit/797e13ad926acfd2a58c34d8df77da45112a818b

Also the TODOs from that commit:
1. We rely here on `ig`. We should build it first -> make dependency. But these are different makefiles
2. We need the path to the `ig` executable we built in 1.
3. The output when using `make build -j` (parallel build) is not that great

Reply:
Thanks for the suggestion. I addresses the TODOs by adding a rule on the top makefile.